spec/expression.dd

```diff
@@ -1266,43 +1266,51 @@ $(GNAME Slice):
 )
 
     $(P $(I PostfixExpression) is evaluated.
-        if $(I PostfixExpression) is an expression of type
-        static array or dynamic array, the special variable $(DOLLAR)
-        is declared and set to be the length of the array.
+        If $(I PostfixExpression) is an expression of static array or
+        dynamic array type, the result of the slice is a dynamic array
+        of the element type of the $(I PostfixExpression).
+    )
+
+    $(P If $(I PostfixExpression) is a $(DDSUBLINK spec/template, variadic-templates, $(I ValueSeq)), then
+        the result of the slice is a new $(I ValueSeq) formed
+        from the upper and lower bounds, which must statically evaluate
+        to integral constants.
+        It is an error if those bounds are out of range.
```

Comment:
```suggestion
        These bounds must be in range.
```

Reply:
Reads a little bit nicer + not a huge fan of language specs mentioning "error" and similar.

Reply:
'must be' is not as clear that this is checked at compile time. (I didn't change the original sentence, just moved it).